### PR TITLE
Lua and Lmod easyconfig using dummy toolchain

### PR DESCRIPTION
**Pull request by pescobar**
There is no Lmod easyconfig using the dummy toolchain so here it is.

I also added the ` --with-static=yes` to the `configopts` in the Lua easyconfig and few other `configopts` which I find useful for the Lmod easyconfig

### CODE REVIEW

File: easybuild/easyconfigs/l/Lmod/Lmod-6.3.eb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+easyblock = 'ConfigureMake'
+
+name = "Lmod"
+version = "6.3"
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/l/Lmod/Lmod-6.3.eb]: recommendation for 6.3.7, several bugfixes landed there!

thanks for the contrib! together with a trick like this, it brings down eb's dependencies from 3 to 2: ;-) 
https://github.com/fgeorgatos/easybuild.experimental/tree/master/users/fgeorgatos/eb_boot

Comment (boegel) [inline easybuild/easyconfigs/l/Lmod/Lmod-6.3.eb]: +1, download from https://github.com/TACC/Lmod/releases

File: easybuild/easyconfigs/l/Lmod/Lmod-6.3.eb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+sources = [SOURCE_TAR_BZ2]
+source_urls = [
+    'http://sourceforge.net/projects/lmod/files/',
+    'http://sourceforge.net/projects/lmod/files/Testing'
+]
+
+builddependencies = [("Lua", "5.1.4-8")]
+
+configopts = ' --with-lua-include=${EBROOTLUA}/include/ --with-mpathSearch=YES --with-caseIndependentSorting=YES --with-pinVersions=YES'
```
Comment (boegel) [inline easybuild/easyconfigs/l/Lmod/Lmod-6.3.eb]: line too long?

**Comment by pescobar** [comment]
should I rename `Lua-5.1.4-8.eb` to something like `Lua-Lmod-5.1.4-8.eb` ?

this is not really the official Lua release but the modified Lua version which Lmod distributes in its website including the required extensions for Lmod...

**Comment by boegel** [comment]
@pescobar maybe use `-Lmod` as version suffix?

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/94eb107b77655fb3520171ef2ca719b3 for a full test report.

**Comment by boegel** [comment]
@pescobar missing dep?

```
gcc -o luac  luac.o print.o liblua.a -lm -Wl,-E -ldl -Wl,-Bstatic -lhistory -lreadline -ltinfo -lncurses  -Wl,-Bdynamic
/usr/bin/ld: cannot find -lhistory 
```

File: easybuild/easyconfigs/l/Lmod/Lmod-6.3.7.eb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+homepage = "http://sourceforge.net/projects/lmod/"
+description = """Lmod is a Lua based module system. Modules allow for dynamic modification
+ of a user's environment under Unix systems. See www.tacc.utexas.edu/tacc-projects/lmod
+ for a complete description. Lmod is a new implementation that easily handles the MODULEPATH
+ Hierarchical problem. It is drop-in replacement for TCL/C modules and reads TCL modulefiles directly."""
+
+toolchain = {'name': 'dummy', 'version': ''}
+
+source_urls = ['https://github.com/TACC/Lmod/archive/']
+sources = ['%(version).tar.gz']
```
Comment (boegel) [inline easybuild/easyconfigs/l/Lmod/Lmod-6.3.7.eb]: @pescobar missing `s` in `%(version)s`

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 3 (3 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/a4dd61b3bcd3b14d84e342ff55b39f06 for a full test report.

**Comment by boegel** [comment]
@pescobar 

```
gcc -o luac  luac.o print.o liblua.a -lm -Wl,-E -ldl -Wl,-Bstatic -lhistory -lreadline -ltinfo -lncurses  -Wl,-Bdynamic
/usr/bin/ld: cannot find -ltinfo  
```

File: easybuild/easyconfigs/l/Lmod/Lmod-6.3.7.eb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+easyblock = 'ConfigureMake'
+
+name = "Lmod"
+version = "6.3.7"
```
Comment (fgeorgatos) [inline easybuild/easyconfigs/l/Lmod/Lmod-6.3.7.eb]: plz, plz, plz. update this to 6.4.1 (argument: `module available` now works... yeah, right)

**Comment by boegel** [comment]
@pescobar an Lmod easyconfig for a newer version is already merged via #3229, shall we close this?

**Comment by pescobar** [comment]
as lua/lmod is already merged I will close this one

**Pull request closed without merging**